### PR TITLE
Nomad Dataset and Devset Update

### PR DESCRIPTION
**Pull request by melo-gonzo**
Updating devset and dataset to include new regression keys" `"reference_energy", "reference_structure", "relative_energy"`. Updated nomad example script to use one of the new keys.

### CODE REVIEW

File: matsciml/datasets/nomad/devset/devset_ids.yml
Comment (laserkelvin) [inline matsciml/datasets/nomad/devset/devset_ids.yml:1]: Why was this file deleted?

**Review by laserkelvin — CHANGES_REQUESTED** [review]

File: matsciml/datasets/nomad/devset/devset_ids.yml
Comment (melo-gonzo) [reply matsciml/datasets/nomad/devset/devset_ids.yml:1]: Those id's no longer reflect what is in the devset (first 200 samples of train lmdb). Also, none of the other datasets have this file so it may be confusing in the future.

**Review by laserkelvin — APPROVED** [review]
LGTM
